include/moq/detail/quic_transport.h

```diff
@@ -214,6 +208,38 @@ namespace qtransport {
                                       uint64_t stream_id,
                                       std::optional<DataContextId> data_ctx_id,
                                       bool is_bidir = false) = 0;
+
+            /**
+             * @brief callback notification on connection metrics sampled
+             *
+             * @details This callback will be called when the connection metrics are sampled per connection
+             *
+             * @param sample_time                    Sample time in microseconds
+             * @param conn_id                        Connection ID for the metrics
+             * @param quic_connection_metrics        Connection specific metrics for sample period
+             */
+            virtual void OnConnectionMetricsSampled([[maybe_unused]] const TimeStampUs sample_time,
+                                                    [[maybe_unused]] const TransportConnId conn_id,
+                                                    [[maybe_unused]] QuicConnectionMetrics quic_connection_metrics)
```

Comment:
```suggestion
                                                    [[maybe_unused]] const QuicConnectionMetrics& quic_connection_metrics)
```

Reply:
since it's a copy, why would it need to be const?

Reply:
changed.